test/integration/ovs/ovs_client_test.go

```diff
@@ -287,6 +287,69 @@ func TestTunnelOptionCsum(t *testing.T) {
 	}
 }
 
+func TestTunnelOptionTunnelPort(t *testing.T) {
+	testCases := map[string]struct {
+		initialTunnelPort int32
+		updatedTunnelPort int32
+	}{
+		"initial zero, kept zero": {
+			initialTunnelPort: 0,
+			updatedTunnelPort: 0,
+		},
+		"initial zero, updated to 8472": {
+			initialTunnelPort: 0,
+			updatedTunnelPort: 8472,
+		},
+		"initial 8472, kept 8473": {
+			initialTunnelPort: 8472,
+			updatedTunnelPort: 8473,
+		},
+		"initial 8472, updated to zero": {
+			initialTunnelPort: 8472,
+			updatedTunnelPort: 0,
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			data := &testData{}
+			data.setup(t)
+			defer data.teardown(t)
+
+			name := "vxlan0"
+			extraOptions := map[string]interface{}{}
+			if testCase.initialTunnelPort != 0 {
+				extraOptions["dst_port"] = strconv.Itoa(int(testCase.initialTunnelPort))
+			}
+			_, err := data.br.CreateTunnelPortExt(name, ovsconfig.VXLANTunnel, ofPortRequest, true, "", "", "", "", extraOptions, nil)
+			require.Nil(t, err, "Error when creating tunnel port")
+			options, err := data.br.GetInterfaceOptions(name)
+			require.Nil(t, err, "Error when getting interface options")
+			if dstPort, ok := options["dst_port"]; ok {
+				actualInitialTunnelPort, _ := strconv.ParseBool(dstPort)
+				require.Equal(t, testCase.initialTunnelPort, actualInitialTunnelPort)
```

Comment:
The check should be executed regardless of whether the got options contains "dst_port" or not, otherwise if `CreateTunnelPortExt` is implemented wrongly that it doesn't actually set the dst_port, there would be no failure.

Reply:
Thanks for the review, updated.